extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -94,6 +118,37 @@ class AmpYoutube extends AMP.BaseElement {
   }
 
   /** @private */
+  getPlayerReadyPromise_() {
+    return new Promise(resolve => {
+      this.getWin().addEventListener('message', event => {
+        const data = JSON.parse(event.data);
+        if (event.origin != 'https://www.youtube.com' ||
+            event.source != this.iframe_.contentWindow) {
+          return;
+        }
+        if (data.event == 'onReady') {
+          resolve(this.iframe_);
+        } else if (data.event == 'infoDelivery') {
```

Comment:
I think one listener is fine. You'd have to move it out of promise creation. That part will get a little less attractive, but everything else will look better. It's not unlikely that we may need to process move events from youtube. So, you can just rename this method `listenToYoutube_()` or something like this and move this promise outside in the `buildCallback` or such.